cpp/memilio/geography/tree.h

```diff
@@ -0,0 +1,253 @@
+/* 
+* Copyright (C) 2020-2025 MEmilio
+*
+* Authors: Kilian Volmer
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef TREE_H
+#define TREE_H
+
+#include <boost/geometry/geometries/multi_polygon.hpp>
+#include <boost/geometry/index/rtree.hpp>
+#include <boost/geometry/strategies/cartesian/buffer_side_straight.hpp>
+#include <boost/geometry/strategies/geographic/buffer_point_circle.hpp>
+#include <boost/geometry/geometry.hpp>
+#include <math.h>
+#include <iterator>
+#include <type_traits>
+#include <vector>
+
+namespace bg   = boost::geometry;
+namespace bgi  = bg::index;
+namespace bgsb = bg::strategy::buffer;
+
+namespace mio
+{
+namespace geo
+{
+using Point = bg::model::point<double, 2, boost::geometry::cs::geographic<boost::geometry::degree>>;
+typedef std::pair<Point, size_t> Node;
+
+template <class Location>
+concept IsSphericalLocation = requires(const Location& loc) {
+    std::is_floating_point_v<decltype(loc.get_latitude())> && std::is_floating_point_v<decltype(loc.get_longitude())>;
+};
+
+template <class Iter>
+concept IsSphericalLocationIterator = std::input_iterator<Iter> && IsSphericalLocation<decltype(*std::declval<Iter>())>;
+
+/**
+ * @brief R-Tree for spatial queries of geographical locations on the sphere
```

Comment:
1. I'd prefer some more information what this is and why we need it here in the code documentation.
3. Please check consistent use of R-Tree vs RTree

Reply:
Ok